api/app/src/routes/helpers/provider-route-helper.ts

```diff
@@ -43,6 +44,9 @@ export async function getProviderDataForType<T>(
       data.push(result.value);
     } else {
       console.error(result.reason);
+      capture.error(result.reason, {
+        extra: { context: `${providers[i]}.${type}` },
```

Comment:
How are we sure we can index `providers` with `i` and it would match what we expect? I don't think this will work.

Looking at this, I suggest we add a `.catch()` to the promise on `requests.push()` on like 32, so we report error based on the context we have there (only remembering to rethrow the Error so `.allSettled()` works properly.

Reply:
The promise array is guaranteed to be in the same order as it was created.

Same thing is done on a line below: https://github.com/metriport/metriport/blob/200e8d83d5924066acc2722f920da7e220f0302f/api/app/src/routes/helpers/provider-route-helper.ts#L53

Would like to avoid modifying the core flow of the code here for a small patch

Reply:
Yeah, the result array of Promise.allSettled is in the same order, you're right. I didn't notice the logic with two arrays (providers and result).

Its fine to send this patch, for sure. But as we work on this code I think we should improve it by making the code more cohesive and minimize dependency on array positions and different arrays/structures having data we expect it to have.